docker/images/n8n-custom/Dockerfile

```diff
@@ -18,7 +18,7 @@ RUN find . -type f -name "*.ts" -o -name "*.js.map" -o -name "*.vue" -o -name "t
 
 # Deploy the `n8n` package into /compiled
 RUN mkdir /compiled
-RUN NODE_ENV=production DOCKER_BUILD=true pnpm --filter=n8n --prod --no-optional deploy /compiled
+RUN NODE_ENV=production DOCKER_BUILD=true pnpm --filter=n8n --prod --no-optional --legacy deploy /compiled
```

Comment:
What does the `--legacy` do? Can we document that

Reply:
TBH, I don't know for sure. All I know is that pnpm 10 changed the default implementation for `pnpm deploy` which breaks with `catalog:` dependencies. Using `--legacy` switched back to the behavior we had on pnpm 9.
There is a discussion [here](https://github.com/orgs/pnpm/discussions/3938).
I'm hoping that before pnpm 11 I would have made sense of this change, and can remove the flag by then.

Reply:
Maybe add the link to the discussion as a comment for future reference

Reply:
created a [ticket](https://linear.app/n8n/issue/CAT-636) for it.